pkg/sonar/componentService.go

```diff
@@ -16,17 +17,34 @@ const EndpointMeasuresComponent = "measures/component"
 type ComponentService struct {
 	Organization string
 	Project      string
+	Branch       string
 	apiClient    *Requester
 }
 
 type SonarCoverage struct {
-	Coverage       float32 `json:"coverage,omitempty"`
-	LineCoverage   float32 `json:"lineCoverage,omitempty"`
-	BranchCoverage float32 `json:"branchCoverage,omitempty"`
+	Coverage          float32 `json:"coverage"`
+	LineCoverage      float32 `json:"lineCoverage"`
+	LinesToCover      int     `json:"linesToCover"`
+	UncoveredLines    int     `json:"uncoveredLines"`
+	BranchCoverage    float32 `json:"branchCoverage"`
+	BranchesToCover   int     `json:"branchesToCover"`
+	UncoveredBranches int     `json:"uncoveredBranches"`
 }
 
-// GetCoverage ...
-func (service *ComponentService) Component(options *sonargo.MeasuresComponentOption) (*sonargo.MeasuresComponentObject, *http.Response, error) {
+type SonarLinesOfCode struct {
+	Total                int                         `json:"total"`
+	LanguageDistribution []SonarLanguageDistribution `json:"languageDistribution,omitempty"`
+}
+
+type SonarLanguageDistribution struct {
+	LanguageKey string `json:"languageKey,omitempty"` // Description:"key of the language as retrieved from sonarqube. All languages (key + name) are available as API https://<sonarqube-instance>/api/languages/list ",ExampleValue:"java,js,web,go"
+	LinesOfCode int    `json:"linesOfCode"`
+}
+
+func (service *ComponentService) Component(options *MeasuresComponentOption) (*sonargo.MeasuresComponentObject, *http.Response, error) {
+	if len(service.Branch) > 0 {
```

Comment:
Your're passing both the `Organization`, `Project` and `Branch` to the service but only overwrite `options.Branch` with `service.Branch` here. Why is that?

Reply:
options does not have Organization and Project, the componentservice does (which was already the case before).